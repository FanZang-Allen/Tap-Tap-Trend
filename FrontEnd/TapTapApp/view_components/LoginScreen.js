import React from 'react';
import {useState} from 'react';
import {
  View,
  Text,
  ImageBackground,
  StyleSheet,
  Button,
  TextInput,
  Alert,
} from 'react-native';
import {user_data, baseUrl, headers, ChangeUser} from '../model/Config';

const LoginScreen = ({navigation}) => {
  const [backText, setBackText] = useState('');
  const [showLogin, setShowLogin] = useState(true);
  const [inputID, setInputID] = useState('');
  const [inputPassword, setInputPassword] = useState('');

  function onChangePassword(newPassword) {
    setInputPassword(newPassword);
  }

  function onChangeID(newID) {
    setInputID(newID);
  }

  function pressBack() {
    setBackText('');
    setShowLogin(true);
  }

  async function pressSignIn() {
    if (inputID === '') {
      Alert.alert('Error', 'Please input a user name', [
        {text: 'OK', onPress: () => console.log('OK Pressed')},
      ]);
    } else if (inputPassword === '') {
      Alert.alert('Error', 'Please input a password', [
        {text: 'OK', onPress: () => console.log('OK Pressed')},
      ]);
    } else {
      try {
        const fetchUrl = baseUrl + `/api/user/info?id=${inputID}`;
        let response = await fetch(fetchUrl, {
          method: 'GET',
        });
        let response_data = await response.json();
        if (response.status !== 200) {
          Alert.alert('Error', 'Connection Error', [
            {text: 'OK', onPress: () => console.log('OK Pressed')},
          ]);
        } else {
          ChangeUser(response_data['Query Result'][0]);
        }
      } catch (e) {
        console.log(e);
      }
      if (user_data.password !== inputPassword) {
        Alert.alert('Error', 'Incorrect uid or password.', [
          {text: 'OK', onPress: () => console.log('OK Pressed')},
        ]);
      } else {
        setInputID('');
        setInputPassword('');
        navigation.navigate('Home');
      }
    }
  }

  async function pressSignUp() {
    if (showLogin === true) {
      setBackText('< Back');
      setShowLogin(false);
      return;
    } else {
      if (inputID === '') {
        Alert.alert('Error', 'Please input a user name', [
          {text: 'OK', onPress: () => console.log('OK Pressed')},
        ]);
      } else if (inputPassword === '') {
        Alert.alert('Error', 'Please input a password', [
          {text: 'OK', onPress: () => console.log('OK Pressed')},
        ]);
      } else {
        try {
          const fetchUrl = baseUrl + '/api/user/add';
          const data = {user_name: inputID, password: inputPassword};
          let response = await fetch(fetchUrl, {
            method: 'POST',
            headers: headers,
            body: JSON.stringify(data),
          });
          let response_data = await response.json();
          if (response.status !== 200) {
            Alert.alert('Error', response_data.Error_message, [
              {text: 'OK', onPress: () => console.log('OK Pressed')},
            ]);
          } else {
            Alert.alert('Success', response_data.Response, [
              {text: 'OK', onPress: () => console.log('OK Pressed')},
            ]);
            setInputID('');
            setInputPassword('');
          }
        } catch (e) {
          console.log(e);
        }
      }
    }
  }

  return (
    <View style={styles.container}>
      <ImageBackground
        source={require('../assets/Background.png')}
        style={styles.background}>
        <Text style={styles.header}>TapTap Trend</Text>
        <View style={styles.input_container}>
          <Text style={styles.back_text} onPress={pressBack}>
            {backText}
          </Text>
          <TextInput
            style={styles.text_input}
            onChangeText={onChangeID}
            value={inputID}
            placeholder={showLogin ? 'UID' : 'UserName'}
            keyboardType={showLogin ? 'numeric' : 'default'}
          />
          <TextInput
            style={styles.text_input}
            onChangeText={onChangePassword}
            value={inputPassword}
            placeholder="Password"
            keyboardType="numeric"
          />
          <View style={styles.margin_container}>
            {showLogin && <Button title="Sign in" onPress={pressSignIn} />}
          </View>
          <View style={styles.margin_container}>
            <Button title="Sign Up" color="#F5A623" onPress={pressSignUp} />
          </View>
        </View>
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  background: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 20,
    paddingBottom: 10,
  },
  header: {
    fontSize: 36,
    color: '#ffffff',
    fontStyle: 'italic',
    fontWeight: 'bold',
    marginBottom: 30,
  },
  back_text: {
    fontSize: 18,
    marginBottom: 10,
  },
  margin_container: {
    marginBottom: 20,
    marginHorizontal: 10,
  },
  text_input: {
    height: 35,
    width: 210,
    borderWidth: 1,
    padding: 10,
    fontWeight: 'bold',
    marginBottom: 20,
    marginHorizontal: 10,
  },
  input_container: {
    width: 270,
    height: 320,
    backgroundColor: '#ffffff',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    borderRadius: 15,
    padding: 20,
  },
  avatar_image: {
    width: '30%',
    height: '20%',
    borderRadius: 100,
  },
  text_container: {
    width: '90%',
    height: '45%',
    justifyContent: 'space-evenly',
  },
  bio_container: {
    width: '90%',
    height: '10%',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'baseline',
  },
  bio_scrollview: {
    flex: 1,
  },
  navigation_container: {
    width: '90%',
    height: '15%',
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  navigation_text: {
    color: '#4A08DB',
    fontSize: 24,
    fontWeight: 'bold',
    textDecorationLine: 'underline',
  },
  text: {
    color: 'white',
    fontSize: 24,
  },
  footer_text: {
    color: 'white',
    fontSize: 16,
  },
});

export default LoginScreen;
