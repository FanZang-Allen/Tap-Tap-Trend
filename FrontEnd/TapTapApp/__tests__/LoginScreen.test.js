import 'react-native';
import React from 'react';
import LoginScreen from '../view_components/LoginScreen';
import renderer from 'react-test-renderer';

test('Login Rendered Page', () => {
  const snap = renderer.create(<LoginScreen />).toJSON();
  expect(snap).toMatchSnapshot();
});
