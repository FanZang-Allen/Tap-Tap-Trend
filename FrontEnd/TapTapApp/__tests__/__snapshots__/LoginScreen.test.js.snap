// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Rendered Page 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 36,
          "fontStyle": "italic",
          "fontWeight": "bold",
          "marginBottom": 30,
        }
      }
    >
      TapTap Trend
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "flexDirection": "column",
          "height": 320,
          "justifyContent": "flex-start",
          "padding": 20,
          "width": 270,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        
      </Text>
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="UID"
        style={
          Object {
            "borderWidth": 1,
            "fontWeight": "bold",
            "height": 35,
            "marginBottom": 20,
            "marginHorizontal": 10,
            "padding": 10,
            "width": 210,
          }
        }
        value=""
      />
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Password"
        style={
          Object {
            "borderWidth": 1,
            "fontWeight": "bold",
            "height": 35,
            "marginBottom": 20,
            "marginHorizontal": 10,
            "padding": 10,
            "width": 210,
          }
        }
        value=""
      />
      <View
        style={
          Object {
            "marginBottom": 20,
            "marginHorizontal": 10,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Sign in
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
            "marginHorizontal": 10,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#F5A623",
                  },
                ]
              }
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
